src/libraries/System.Runtime/tests/System/GCTests.cs

```diff
@@ -1058,11 +1058,52 @@ private static void AllocateArrayTooLarge()
             Assert.Throws<OutOfMemoryException>(() => GC.AllocateUninitializedArray<double>(int.MaxValue, pinned: true));
         }
 
-        [Fact]
-        private static void AllocateArrayRefType()
+        struct EmbeddedValueType<T>
+        {
+            unsafe fixed byte _[7];
```

Comment:
Is there a specific reason for this? It looks like trying to fiddle with offsets and alignment. I tend to avoid these kind of mechanisms in tests without a comment as to the why.

Reply:
It was a simple way of increasing the size of the struct, making it less trivially laid out and requiring padding bits - to add a bit of confidence in we're not just overwriting the first set of bytes in a memory location.

We could add a comment or just remove it, it's not required.

Reply:
I think a non-generic struct like this would be sufficient for what you are trying to test here
```
private struct StructType
{
    internal object _;
    internal string _value;
    internal int _;
}
```